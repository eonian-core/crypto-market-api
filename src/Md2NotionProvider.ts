import {markdownToBlocks, markdownToRichText} from '@tryfabric/martian';
import type { BlocksOptions, RichTextOptions } from '@tryfabric/martian/build/src/parser/internal';


export class Md2NotionProvider {
  async toBlocks(markdown: string, options?: BlocksOptions) {
    return markdownToBlocks(markdown, options);
  }

  async toRichText(markdown: string, options?: RichTextOptions) {
    return markdownToRichText(markdown, options);
  }
}