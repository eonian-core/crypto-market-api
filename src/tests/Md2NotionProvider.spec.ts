import { Md2NotionProvider } from '../Md2NotionProvider';
import { markdownToBlocks, markdownToRichText } from '@tryfabric/martian';

const markdown = `
hello _world_ 
*** 
## heading2
* [x] todo
`;

describe('Md2NotionProvider', () => {
  let provider: Md2NotionProvider;

  beforeEach(() => {
    provider = new Md2NotionProvider();
  });

  describe('toBlocks method', () => {
    it('should convert markdown to blocks correctly', async () => {
      const result = await provider.toBlocks(markdown);
      
      expect(result).toMatchSnapshot();
    });
  });

  describe('toRichText method', () => {
    it('should convert markdown to rich text correctly', async () => {
      const text = '**Hello _world_**';

      const result = await provider.toRichText(text);
      
      expect(result).toMatchSnapshot();
    });
  });
});