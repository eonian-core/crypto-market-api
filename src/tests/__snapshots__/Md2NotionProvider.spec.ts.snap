// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Md2NotionProvider toBlocks method should convert markdown to blocks correctly 1`] = `
[
  {
    "object": "block",
    "paragraph": {
      "rich_text": [
        {
          "annotations": {
            "bold": false,
            "code": false,
            "color": "default",
            "italic": false,
            "strikethrough": false,
            "underline": false,
          },
          "text": {
            "content": "hello ",
            "link": undefined,
          },
          "type": "text",
        },
        {
          "annotations": {
            "bold": false,
            "code": false,
            "color": "default",
            "italic": true,
            "strikethrough": false,
            "underline": false,
          },
          "text": {
            "content": "world",
            "link": undefined,
          },
          "type": "text",
        },
      ],
    },
    "type": "paragraph",
  },
  {
    "heading_2": {
      "rich_text": [
        {
          "annotations": {
            "bold": false,
            "code": false,
            "color": "default",
            "italic": false,
            "strikethrough": false,
            "underline": false,
          },
          "text": {
            "content": "heading2",
            "link": undefined,
          },
          "type": "text",
        },
      ],
    },
    "object": "block",
    "type": "heading_2",
  },
  {
    "object": "block",
    "to_do": {
      "checked": true,
      "children": undefined,
      "rich_text": [
        {
          "annotations": {
            "bold": false,
            "code": false,
            "color": "default",
            "italic": false,
            "strikethrough": false,
            "underline": false,
          },
          "text": {
            "content": "todo",
            "link": undefined,
          },
          "type": "text",
        },
      ],
    },
    "type": "to_do",
  },
]
`;

exports[`Md2NotionProvider toRichText method should convert markdown to rich text correctly 1`] = `
[
  {
    "annotations": {
      "bold": true,
      "code": false,
      "color": "default",
      "italic": false,
      "strikethrough": false,
      "underline": false,
    },
    "text": {
      "content": "Hello ",
      "link": undefined,
    },
    "type": "text",
  },
  {
    "annotations": {
      "bold": true,
      "code": false,
      "color": "default",
      "italic": true,
      "strikethrough": false,
      "underline": false,
    },
    "text": {
      "content": "world",
      "link": undefined,
    },
    "type": "text",
  },
]
`;
